### PR TITLE
Remove all CodeClimate references

### PR DESCRIPTION
**Pull request by brbrr**
#### Changes proposed in this Pull Request:

* Remove CodeClimate

#### Jetpack product discussion

p9dueE-2ND-p2 

#### Does this pull request change what data or activity we track or use?

n/a

#### Testing instructions:

* Check for "CodeClimate" mentions in the codebase
* Make sure that CodeClimate is pending in checks list

### CODE REVIEW

**Review by anomiex — COMMENTED** [review]
Update [docs/monorepo.md](https://github.com/Automattic/jetpack/blob/2745359/docs/monorepo.md#code-coverage) to link to https://docs.codecov.io/docs/supported-report-formats for the supported formats and remove the bits about file naming restrictions?

File: docs/monorepo.md [hunk 1]
```diff
@@ -128,8 +137,8 @@ The Jetpack Monorepo includes GitHub actions to run a number of CI checks on all
 
 Tests for a project are only run for a PR if changes are made to the project or its dependencies. Dependencies may be specified as:
 
-* For Composer packages included in the monorepo, via `.require' and `.require-dev` in `composer.json`.
-* For any other dependencies, via `.extra.dependencies` in `composer.json`.
+- For Composer packages included in the monorepo, via `.require' and`.require-dev`in`composer.json`.
```
Comment (anomiex) [inline docs/monorepo.md:140]: This line was broken already (thanks to the single-quote rather than backquote after `.require`), but you broke it more by removing spaces.

File: docs/monorepo.md [hunk 2]
```diff
@@ -211,7 +217,7 @@ There's no need to be concerned about collisions with other projects' coverage f
 
 Most projects in the monorepo should have a mirror repository holding a built version of the project, ready for deployment. Follow these steps to create the mirror repo and configure the monorepo tooling to push to it.
 
-1. Create the mirror repo on GitHub. It will most likely be named like "https://github.com/Automattic/jetpack-_something_".
+1. Create the mirror repo on GitHub. It will most likely be named like "<https://github.com/Automattic/jetpack>-_something_".
```
Comment (anomiex) [inline docs/monorepo.md:220]: On the other hand, you kind of fixed this line. But an even better fix would be to disable the link entirely, maybe something like
```suggestion
1. Create the mirror repo on GitHub. It will most likely be named like "https://<span></span>github.com/Automattic/jetpack-_something_".
```
Same on original-side line 110. But maybe we should leave all that cleanup for a separate PR.

**Review by anomiex — COMMENTED** [review]
Why are you changing the formatting of all the bulleted lists in monorepo.md? It should be a small change, not one that touches 60-some lines.

You also managed to somehow break the formatting in the "Project structure" section, compare [master](https://github.com/Automattic/jetpack/blob/master/docs/monorepo.md#project-structure) with [your PR](https://github.com/Automattic/jetpack/blob/d0593b31bedbb86e8f19769910a8e6e973227ae2/docs/monorepo.md#project-structure).

File: docs/monorepo.md (hunk 1 above)
Comment (brbrr) [reply docs/monorepo.md:140]: 🤦 It's an VScode's automated Markdown linting. I'll revert these changes

**Review by anomiex — APPROVED** [review]
